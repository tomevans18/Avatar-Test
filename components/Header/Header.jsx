import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

import ContentWrapper from '../ContentWrapper';

import StyledHeader, { SkipToMain, LogoWrapper, Logo } from './Header.style';

const propTypes = {
  mainId: PropTypes.string.isRequired,
};

const Header = ({ mainId }) => (
  <StyledHeader>
    <SkipToMain href={mainId}>Skip to main content</SkipToMain>
    <ContentWrapper flex>
      <LogoWrapper>
        <Link href="/">
          <Logo aria-label="Home">
            <img className="site-logo" src="https://via.placeholder.com/175x90" alt="Logo" />
          </Logo>
        </Link>
      </LogoWrapper>
    </ContentWrapper>
  </StyledHeader>
);

Header.propTypes = propTypes;

export default Header;
