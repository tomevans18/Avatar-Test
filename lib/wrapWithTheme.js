import { ThemeConsumer } from 'styled-components';
import theme from '../styles/theme';

export default (method, children, options, themeObject = theme) => {
  // eslint-disable-next-line no-underscore-dangle
  ThemeConsumer._currentValue = themeObject;
  return method(children, options);
};
