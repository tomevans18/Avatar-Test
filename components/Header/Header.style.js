import styled from 'styled-components';

import { multiply } from '../../styles/lib/calculate';

export default styled.header`
  display: flex;
  flex-direction: column;
  border-bottom: 1px solid ${p => p.theme.color.grey};
`;

export const SkipToMain = styled.a`
  left: -999px;
  position: absolute;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  z-index: -999;
  &:focus,
  &:active {
    left: auto;
    top: auto;
    overflow: auto;
    margin: 1rem 35%;
    z-index: 999;
    width: auto;
    height: auto;
    border: none;
    padding: ${p => `
      ${multiply(p.theme.component.padding, 0.8)}
      ${multiply(p.theme.component.padding, 1.15)}
      ${multiply(p.theme.component.padding, 0.8)}
      ${multiply(p.theme.component.padding, 1.15)}
    `};
    text-align: center;
    cursor: pointer;
  }
`;

export const LogoWrapper = styled.div`
  max-width: 200px;
`;

export const Logo = styled.a`
  display: block;
  padding: 0.75rem 0 0.5rem;
  max-width: 70vw;
  cursor: pointer;

  img {
    min-height: 65px;
    max-height: 10vh;
  }
`;

export const LoginBtn = styled.button``;
