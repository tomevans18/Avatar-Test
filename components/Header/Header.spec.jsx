import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import wrapWithTheme from '../../lib/wrapWithTheme';

import Header from './Header';

describe('Header', () => {
  it('should render as expected ', () => {
    const wrapper = wrapWithTheme(mount, <Header />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
