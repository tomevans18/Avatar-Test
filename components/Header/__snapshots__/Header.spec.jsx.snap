// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render as expected  1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + 1.25em);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #ffffff;
  color: #ffffff;
  border-bottom: 1px solid #ccc;
}

.c1 {
  left: -999px;
  position: absolute;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  z-index: -999;
}

.c1:focus,
.c1:active {
  left: auto;
  top: auto;
  overflow: auto;
  margin: 1rem 35%;
  z-index: 999;
  width: auto;
  height: auto;
  border: none;
  padding: 0in 0in 0in 0in;
  text-align: center;
  cursor: pointer;
}

.c3 {
  max-width: 200px;
}

.c4 {
  display: block;
  padding: 0.75rem 0 0.375rem;
  max-width: 70vw;
  cursor: pointer;
}

@supports (padding:max(0px)) {
  .c2 {
    padding: 0 max(1.25em,env(safe-area-inset-left));
  }
}

@media (min-width:769px) {
  .c0 {
    border-bottom: none;
  }
}

<Header>
  <styled.header>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  background: ",
              [Function],
              ";
  color: ",
              [Function],
              ";
  border-bottom: 1px solid ",
              [Function],
              ";

  ",
              "
    @media (min-width: ",
              "769",
              "px) {
      ",
              "
    border-bottom: none;
  ",
              "
    }
    ",
              "
  ",
              ";
",
            ],
          },
          "displayName": "styled.header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <header
        className="c0"
      >
        <styled.a>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  left: -999px;
  position: absolute;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  z-index: -999;
  &:focus,
  &:active {
    left: auto;
    top: auto;
    overflow: auto;
    margin: 1rem 35%;
    z-index: 999;
    width: auto;
    height: auto;
    border: none;
    padding: ",
                    [Function],
                    ";
    text-align: center;
    cursor: pointer;
  }
",
                  ],
                },
                "displayName": "styled.a",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <a
              className="c1"
            >
              Skip to main content
            </a>
          </StyledComponent>
        </styled.a>
        <ContentWrapper
          flex={true}
        >
          <styled.div
            flex={true}
          >
            <StyledComponent
              flex={true}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "
  ",
                      [Function],
                      " position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + ",
                      [Function],
                      ");

  @supports (padding: max(0px)) {
    padding: 0 max(",
                      [Function],
                      ", env(safe-area-inset-left));
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c2"
              >
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": true,
                          "lastClassName": "c3",
                          "rules": Array [
                            "
  max-width: 200px;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="c3"
                    >
                      <Component
                        href="/"
                      >
                        <styled.a
                          aria-label="Home"
                        >
                          <StyledComponent
                            aria-label="Home"
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-ifAKCX",
                                  "isStatic": true,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    "
  display: block;
  padding: 0.75rem 0 0.375rem;
  max-width: 70vw;
  cursor: pointer;
",
                                  ],
                                },
                                "displayName": "styled.a",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-ifAKCX",
                                "target": "a",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <a
                              aria-label="Home"
                              className="c4"
                            >
                              <img
                                alt="Logo"
                                className="site-logo"
                                src="https://via.placeholder.com/175x90"
                              />
                            </a>
                          </StyledComponent>
                        </styled.a>
                      </Component>
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </StyledComponent>
          </styled.div>
        </ContentWrapper>
      </header>
    </StyledComponent>
  </styled.header>
</Header>
`;
